### DISCUSSION TITLE
Use 7-Zip instead of p7zip

### DISCUSSION THREAD

**Discussion by Niaw** [xthursdayx/yacreaderlibrary-server-docker#18] [Ideas]
Hello,

The last version of 7-Zip (v21.07, 2021-12-26) comes now with a Linux version (arm, arm64, x86, x86-64).

I was wondering if this version could replace the old p7zip (v16.02, 2016-07-14) in the Docker container?

Thanks a lot for your work.

**Comment by xthursdayx** [comment]
Thanks for bringing this up. One thing issue is that YACReaderLibraryServer itself cannot currently be built with this newer version of 7-zip. [This README](https://github.com/YACReader/yacreader/blob/develop/compressed_archive/README_7zip.txt) only mentions bundling with 7-Zip version 18.05 (for Windows only) or p7zip version 16.02 (for Linux/MacOSX), while the [INSTALL.md guide](https://github.com/YACReader/yacreader/blob/develop/INSTALL.md#7zip) mentions that the YACRLibraryServer backend requires those specific versions of 7zip and p7zip. Also, this (closed) issue https://github.com/YACReader/yacreader/issues/52 from May 2019 suggests updating the 7-zip sdk to v19 and p7zip to v16.02, however only the latter seems to have been done. In the comments the primary dev notes that "keeping 7zip up to date is a nasty chore," but that updates to Qt7zip may help to facilitate updates to the 7zip wrapper. There is more information about updating p7zip and 7zip integration in this (also closed) issue https://github.com/YACReader/yacreader/issues/274 from August 2021, though this is from before 7-zip had a version for Linux.

I will either comment on issue 274 or start a new issue now that the Linux version of 7zip has been released to see if there are plans to update the backend. Once I figure that out I'll see about updating the docker image. Cheers!

**Comment by Niaw** [comment]
Thank you very much for the detailed explanation.
